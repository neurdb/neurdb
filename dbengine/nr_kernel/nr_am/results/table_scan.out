DROP TABLE IF EXISTS x;
NOTICE:  table "x" does not exist, skipping
DROP TABLE IF EXISTS y;
NOTICE:  table "y" does not exist, skipping
CREATE TABLE x(a INT, b text, c text, PRIMARY KEY (a, b)) USING nram;
INFO:  [NRAM] calling function nram_relation_set_new_filelocator
INFO:  [NRAM] calling function nram_relation_needs_toast_table
INFO:  [NRAM] calling function nram_estimate_rel_size
INFO:  [NRAM] calling function nram_index_build_range_scan
CREATE TABLE y(a INT, b text, PRIMARY KEY (a)) USING nram;
INFO:  [NRAM] calling function nram_relation_set_new_filelocator
INFO:  [NRAM] calling function nram_relation_needs_toast_table
INFO:  [NRAM] calling function nram_estimate_rel_size
INFO:  [NRAM] calling function nram_index_build_range_scan
-- Test insert and table scan.
INSERT INTO x VALUES (1, 'k1', 'v1');
INFO:  [NRAM] calling function nram_slot_callbacks
INFO:  [NRAM] calling function nram_tuple_insert
INFO:  [NRAM] calling function nram_insert
INFO:  [NRAM] [nram_insert:239] nram tid generated as 42430
INFO:  [NRAM] calling function rocksengine_open
INFO:  [NRAM] [find_write_set:110] write_set=(nil) length=0
INFO:  [NRAM] [find_write_set:110] write_set=0x55b9997e45e8 length=1
INFO:  [NRAM] [nram_insert:247] ADD! key = <16411:42430>
INFO:  [NRAM] [nram_xact_callback:189] The callback is triggered on event PRE_COMMIT
INFO:  [NRAM] [nram_xact_callback:205] Pre-validation add write locks.
INFO:  [NRAM] [nram_xact_callback:230] Post-validation release write locks.
INFO:  [NRAM] [nram_xact_callback:238] Validation succeed.
INFO:  [NRAM] [nram_xact_callback:189] The callback is triggered on event COMMIT
INFO:  [NRAM] [nram_xact_callback:250] the transaction 766 is committed
SELECT * from x;
INFO:  [NRAM] calling function nram_estimate_rel_size
INFO:  [NRAM] calling function nram_slot_callbacks
INFO:  [NRAM] calling function nram_beginscan
INFO:  [NRAM] calling function nram_getnextslot
server closed the connection unexpectedly
	This probably means the server terminated abnormally
	before or while processing the request.
connection to server was lost
