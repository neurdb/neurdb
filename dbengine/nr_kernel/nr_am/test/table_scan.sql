CREATE TABLE x(a INT, b text, c text, PRIMARY KEY (a, b)) USING nram;
CREATE TABLE y(a INT, b text, PRIMARY KEY (a)) USING nram;

-- Test insert and table scan.
INSERT INTO x VALUES (1, 'k1', 'v1');
SELECT * from x;
INSERT INTO x VALUES (2, 'k2', 'v2');
SELECT * from x;
-- Test different tables.
INSERT INTO y VALUES (1, 'z1');
INSERT INTO y VALUES (2, 'z2');
SELECT * from x;
SELECT * from y;
-- Drop all.
DROP TABLE x;
DROP TABLE y;
