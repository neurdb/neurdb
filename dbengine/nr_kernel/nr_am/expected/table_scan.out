DROP TABLE IF EXISTS x;
NOTICE:  table "x" does not exist, skipping
DROP TABLE IF EXISTS y;
NOTICE:  table "y" does not exist, skipping
CREATE TABLE x(a INT, b text, c text, PRIMARY KEY (a, b)) USING nram;
CREATE TABLE y(a INT, b text, PRIMARY KEY (a)) USING nram;
-- Test insert and table scan.
INSERT INTO x VALUES (1, 'k1', 'v1');
SELECT * from x;
 a | b  | c  
---+----+----
 1 | k1 | v1
(1 row)

INSERT INTO x VALUES (2, 'k2', 'v2');
SELECT * from x;
 a | b  | c  
---+----+----
 1 | k1 | v1
 2 | k2 | v2
(2 rows)

-- Test different tables.
INSERT INTO y VALUES (1, 'z1');
INSERT INTO y VALUES (2, 'z2');
SELECT * from x;
 a | b  | c  
---+----+----
 1 | k1 | v1
 2 | k2 | v2
(2 rows)

SELECT * from y;
 a | b  
---+----
 1 | z1
 2 | z2
(2 rows)

-- Drop all.
DROP TABLE x;
DROP TABLE y;
